src/components/SearchBox.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import Urls from '../constants/Urls';
+
+class SearchBox extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			searchTerm: '',
+		};
+	
+		this.handleTextChange = this.handleTextChange.bind(this);
+		this.handleSearchClick = this.handleSearchClick.bind(this);
+	}
+
+	handleTextChange(changeEvent) {
+		this.setState({ searchTerm: changeEvent.target.value.trim() })
+	}
+
+	handleSearchClick() {
+		if (this.state.searchTerm.trim().length === 0) {
+			return;
+		}
+
+		if (!this.props.onResults) {
+			console.error('Results are unhandled.');
+			return;
+		}
+
+		fetch(`${Urls.search}/${this.state.searchTerm}`)
+			.then(results => results.json())
+			.then(jsonResults => this.props.onResults(jsonResults))
```

Comment:
to me it seems like the results should be represented in the state of the app

Reply:
They do become state in the component this one is nested in. I didn't see any reason to do it here.